Cue/app/tabs/library/play/PlayDeckView.js

```diff
@@ -0,0 +1,487 @@
+    // to still capture taps
+    let { dx, dy } = gestureState
+    return Math.abs(dx) > 5 || Math.abs(dy) > 5
+  }
+
+  _onPanResponderGrant = (event, gestureState) => {
+    // Pan event has been captured - show the indicators
+    this._showIndicators().start()
+  }
+
+  _onPanResponderMove = (event, gestureState) => {
+    // "Snap" the pan to a movement along a single axis
+    let snapDirection = Math.abs(gestureState.dx) >= Math.abs(gestureState.dy) ? 'x' : 'y'
+
+    let snappedGestureState = {
+      ...gestureState,
+      dx: snapDirection === 'x' ? gestureState.dx : 0,
+      dy: snapDirection === 'y' ? gestureState.dy : 0,
+    }
+
+    Animated.event([null, {
+      dx: this.state.cardXY.x,
+      dy: this.state.cardXY.y,
+    }])(event, snappedGestureState)
+
+    let detectedAction = this._triggeredActionForGesture(gestureState)
+
+    if (detectedAction !== this.state.triggeredAction) {
+      this.setState({
+        ...this.state,
+        triggeredAction: detectedAction
+      })
+    }
+  }
+
+  _onPanResponderRelease = (event, gestureState) => {
+    if (this.state.triggeredAction === 'none') {
+      Animated.sequence([
+        Animated.spring(
+          this.state.cardXY, {
+            toValue: { x: 0, y: 0 },
+            ...baseSpringConfig
+          }
+        ),
+        this._hideIndicators()
+      ]).start()
+    } else {
+      this._commitAction(this.state.triggeredAction)
+    }
+  }
+
+  _onPanResponderTerminate = (event, gestureState) => {
+    return this._onPanResponderRelease(event, gestureState)
+  }
+
+
+  /* ==================== Floating Action Indicator ==================== */
+
+  _showIndicatorConfig = {
+    toValue: 1,
+    duration: DURATION_STANDARD,
+  }
+
+  _hideIndicatorConfig = {
+    toValue: 0,
+    duration: DURATION_STANDARD,
+  }
+
+  _showOnlyIndicator = (indicator: TriggeredAction) => {
+    return this._setIndicatorOpacity(
+      indicator === 'flag' ? this._showIndicatorConfig : this._hideIndicatorConfig,
+      indicator === 'next' ? this._showIndicatorConfig : this._hideIndicatorConfig,
+      indicator === 'back' ? this._showIndicatorConfig : this._hideIndicatorConfig)
+  }
+
+  _showIndicators = () => {
+    return this._setIndicatorOpacity(
+      this._showIndicatorConfig,
+      this._showIndicatorConfig,
+      this._showIndicatorConfig)
+  }
+
+  _hideIndicators = () => {
+    return this._setIndicatorOpacity(
+      this._hideIndicatorConfig,
+      this._hideIndicatorConfig,
+      this._hideIndicatorConfig)
+  }
+
+  _setIndicatorOpacity = (flagConfig, nextConfig, backConfig) => {
+    return Animated.parallel([
+      Animated.timing(this.state.flagOpacity, flagConfig),
+      Animated.timing(this.state.nextOpacity, nextConfig),
+      Animated.timing(this.state.backOpacity, backConfig)
+    ])
+  }
+
+  _setUpIndicatorAnimation = () => {
+    Animated.spring(
+      this.state.flagY, {
+        toValue: this.state.cardXY.y.interpolate({
```

Comment:
For the flag indicator, we animate its movement along the Y axis by having its `bottom` style value use a spring animation to track the y-value of the card's position. (The position of the card itself is controlled using `this.state.cardXY` which is updated by the `PanResponder` attached to the card view — see `_onPanResponderMove` and `_onPanResponderRelease`.) Since a 1:1 correspondence between them would mean that the indicator could be permanently obscured by the card, we use `interpolate` to reduce the tracking to 0.4:1 so the indicator "lags" behind the card as the card moves towards the top of the screen. The `inputRange` and `outputRange` are set up like so:

```
              ---*---
inputRange:  [ -1, 0, 1]
outputRange: [0.4, 0, 0]
                  --**--
```
\* Since the y-value of the card decreases as the card is moved upwards, but we want the `bottom` value to increase, we map a decrease of 1 in the card y-value to an increase of 0.4 in the `bottom` style value.
\** This range is used for extrapolation so that if the card moves down, the `bottom` value never decreases below 0, ensuring that the indicator never slides off the screen.

A similar strategy is used for the next and back indicators, which track the x-value of `this.state.cardXY`  instead.